unreleased/issue-12423.yaml

```diff
@@ -0,0 +1,16 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: enhancement
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: cumulativetodeltaprocessor
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Cumulative Histogram metrics will now be converted to delta temporality and are no longer skipped.
+
+# One or more tracking issues related to the change
+issues: [12423]
+
+# (Optional) One or more lines of additional information to render under the primary note.
+# These lines will be padded with 2 spaces and then inserted directly into the document.
+# Use pipe (|) for multiline entries.
+subtext: This means that consumers who currently rely on Histograms being skipped would need to ensure they are excluded via config.
```

Comment:
I think as an "ease-in" measure, we shouldn't convert histograms by default just yet.  Some users may be relying on the fact that the processor is skipping histograms because they want their histograms to be cumulative.  Let's add a new config option to enable histogram conversion support.  If enabled, histograms will be subjected to all the filtering logic and then converted if they pass.  If disabled, no histograms will be converted.

Reply:
That makes sense. We could possibly do this more generically as a list of enabled metric types. So it would default to `["sum"]`, but `"histogram"` could be added as well. Just so that this could be extended to other metric types in future without adding bespoke new config options for those.

Reply:
Thinking about it some more I think a [feature gate](https://github.com/open-telemetry/opentelemetry-collector/tree/main/service/featuregate) is a better solution that a temporary config option.  Feature gates allow users to enable features without us having to maintain a new config option and then deprecate it in the future.  

Here is an example of how the hostmetrics receiver is using feature gates to handle some changes to the metrics it collects:
- https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/hostmetricsreceiver#feature-gate-configurations
- https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/12105

We'd define a timeline like the hostmetrics receiver and then eventually remove the feature gate, making converting histograms a default functionality.

Reply:
This makes a lot of sense to me. I committed an attempt at adding a feature flag - is this the right way to do it? Does any other part of the codebase need to know about the flag?

Reply:
This looks like a proper implementation to me.  Were you able to confirm locally that the feature flag works as expected?

Reply:
In the process of doing that now 🙂

Reply:
Awesome!